network/src/protocols/mod.rs

```diff
@@ -33,17 +34,53 @@ pub type BoxedFutureTask = Pin<Box<dyn Future<Output = ()> + 'static + Send>>;
 
 use crate::{
     compress::{compress, decompress},
-    network::disconnect_with_message,
+    network::{async_disconnect_with_message, disconnect_with_message},
     Behaviour, Error, NetworkState, Peer, ProtocolVersion,
 };
 
 /// Abstract protocol context
+#[async_trait]
 pub trait CKBProtocolContext: Send {
     /// Set notify to tentacle
     // Interact with underlying p2p service
-    fn set_notify(&self, interval: Duration, token: u64) -> Result<(), Error>;
+    async fn set_notify(&self, interval: Duration, token: u64) -> Result<(), Error>;
     /// Remove notify
-    fn remove_notify(&self, token: u64) -> Result<(), Error>;
+    async fn remove_notify(&self, token: u64) -> Result<(), Error>;
+    /// Send message through quick queue
+    async fn async_quick_send_message(
+        &self,
+        proto_id: ProtocolId,
+        peer_index: PeerIndex,
+        data: Bytes,
+    ) -> Result<(), Error>;
+    /// Send message through quick queue
+    async fn async_quick_send_message_to(
```

Comment:
I cannot find the usage of these `async_xxx` fn in this PR, please describe the specific use.

Reply:
This is to give the context rewrapped with ckb-network the ability to use async-await. The current sync and relayer protocols are left as they are because of the complexity of locks and states, but that doesn't mean other dependent implementations can't use them.

Whether you can use it and whether you use it are two different things. In an environment where the context can use await, it would be strange not to provide a corresponding method.

There is a difference in the behavior of synchronous and asynchronous methods.
1. when a synchronous method encounters a full queue, it returns a would block error after trying to send and exits directly
2. When an asynchronous method encounters a full queue, it enters a pending state and waits for the next wake